### PR TITLE
Typos: `test/IRGen`

### PR DESCRIPTION
**Pull request by Sajjon**
Fix typos in `test/IRGen`

This is one batch of many PRs fixing typos, [see the tracking issue.](https://github.com/swiftlang/swift/issues/75022)

### CODE REVIEW

File: test/IRGen/weak_import_availability.swift [hunk 1]
```diff
@@ -115,14 +115,14 @@ public func useConditionallyAvailableMethod(s: ConditionallyAvailableStruct) {
 
 @available(macOS, unavailable)
 public func useUnavailableStruct() {
-  blackHole(UnvailableStruct.self)
+  blackHole(UnavailableStruct.self)
 }
 
-// CHECK-LABEL: declare extern_weak swiftcc %swift.metadata_response @"$s31weak_import_availability_helper16UnvailableStructVMa"(i64)
+// CHECK-LABEL: declare extern_weak swiftcc %swift.metadata_response @"$s31weak_import_availability_helper16UnavailableStructVMa"(i64)
```
Comment (AnthonyLatsis) [inline test/IRGen/weak_import_availability.swift:121]:
```suggestion
// CHECK-LABEL: declare extern_weak swiftcc %swift.metadata_response @"$s31weak_import_availability_helper17UnavailableStructVMa"(i64)
```

File: test/IRGen/ptrauth-dynamic_replaceable.sil [hunk 2]
```diff
@@ -45,7 +45,7 @@
 // CHECK-NEXT:   ret void
 // CHECK-NEXT: }
 
-sil [dynamically_replacable] @test_dynamically_replaceable : $@convention(thin) () -> () {
+sil [dynamically_replaceable] @test_dynamically_replaceable : $@convention(thin) () -> () {
```
Comment (AnthonyLatsis) [inline test/IRGen/ptrauth-dynamic_replaceable.sil:48]: Renaming this attribute is a cross-component change, so you may want to leave this out for now.

File: test/IRGen/weak_import_availability.swift [hunk 3]
```diff
@@ -115,14 +115,14 @@ public func useConditionallyAvailableMethod(s: ConditionallyAvailableStruct) {
...
 
 @available(macOS, unavailable)
-public func useUnavailableMethod(s: UnvailableStruct) {
+public func useUnavailableMethod(s: UnavailableStruct) {
   s.unavailableMethod()
 }
 
-// CHECK-LABEL: declare extern_weak swiftcc void @"$s31weak_import_availability_helper16UnvailableStructV17unavailableMethodyyF"(ptr noalias swiftself)
+// CHECK-LABEL: declare extern_weak swiftcc void @"$s31weak_import_availability_helper16UnavailableStructV17unavailableMethodyyF"(ptr noalias swiftself)
```
Comment (AnthonyLatsis) [inline test/IRGen/weak_import_availability.swift:128]:
```suggestion
// CHECK-LABEL: declare extern_weak swiftcc void @"$s31weak_import_availability_helper17UnavailableStructV17unavailableMethodyyF"(ptr noalias swiftself)
```

File: test/IRGen/ptrauth-dynamic_replaceable.sil (hunk 2 above)
Comment (Sajjon) [reply test/IRGen/ptrauth-dynamic_replaceable.sil]: Reverted

**Comment by Sajjon** [comment]
@AnthonyLatsis all suggestions fixed :)

File: test/IRGen/dynamic_replaceable.sil [hunk 4]
```diff
@@ -39,7 +39,7 @@
 // CHECK-NEXT:   ret void
 // CHECK-NEXT: }
 
-sil [dynamically_replacable] @test_dynamically_replaceable : $@convention(thin) () -> () {
+sil [dynamically_replaceable] @test_dynamically_replaceable : $@convention(thin) () -> () {
```
Comment (AnthonyLatsis) [inline test/IRGen/dynamic_replaceable.sil:42]: Ditto.

Comment (Sajjon) [reply test/IRGen/dynamic_replaceable.sil]: reverted

**Comment by Sajjon** [comment]
@AnthonyLatsis I've squashed and rebased on main, and I think applied suggested changes?

Anything more I need to do to get this merge ready?

**Comment by AnthonyLatsis** [comment]
@swift-ci please smoke test

**Review by AnthonyLatsis — APPROVED** [review]
LGTM, thanks. Not a code owner of this area, but I am going to assume I have their trust in my judgement in this sort of case.

**Pull request merged**